src/containers/toputilizers/containers/TopUtilizersResultsContainer.js

```diff
@@ -55,6 +57,90 @@ class TopUtilizersResultsContainer extends React.Component {
     });
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.results.size !== this.state.resultsWithCounts.size) {
+      this.setState({
+        resultsWithCounts: nextProps.results
+      });
+    }
+    if (this.props.isGettingNeighbors && !nextProps.isGettingNeighbors) {
+      this.countEdgeTypesPerEntity(nextProps);
```

Comment:
@solomontang , one thing to think about is invoking `this.setState()` multiple times within `componentWillReceiveProps()`. I'm not sure what the behavior is, but it could be bad if both calls to `this.setState()` end up calling `render()`. If I was doing this, I'd probably have `countEdgeTypesPerEntity()` return the new data, and I'd call `this.setState()` once.

Reply:
I believe calls to setState within `componentWillReceiveProps()` will not trigger multiple renders. 

[https://stackoverflow.com/a/44829336](https://stackoverflow.com/a/44829336)

However, I do believe the lifecycle hook is being phased out in favor of `getStateDerivedFromProps()` in React 17.

[https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)